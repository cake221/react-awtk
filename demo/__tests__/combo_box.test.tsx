import "../combo_box"

describe("test edit", ()=>{

  it("toMatchSnapshot", ()=>{
    expect(__AwtkSnapshot).toMatchSnapshot()
  })
})
