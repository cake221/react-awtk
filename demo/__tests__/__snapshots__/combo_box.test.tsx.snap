// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test edit toMatchSnapshot 1`] = `
Array [
  "getRootHostContext",
  "getChildHostContext",
  "shouldSetTextContent",
  "getChildHostContext",
  "shouldSetTextContent",
  "createInstance",
  "funtion  combo_box_create( parent, x, y, w, h ) :null, 0, 0, 0, 0",
  "funtion  widget_set_self_layout_params( widget, x, y, w, h ) :undefined, 10, 10, 168, 30",
  "funtion  widget_set_name( widget, name ) :undefined, color",
  "funtion  combo_box_set_options( widget, options ) :undefined, Red;Green;Blue",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_VALUE_CHANGED, function (evt) {
        var target = src_1.TEvent.cast(evt).target;
        var instance = src_1.TComboBox.cast(target);
        console.log(instance.name + ' changed: ' + (instance.value));
        return TRet.OK;
    }, null",
  "finalizeInitialChildren",
  "getChildHostContext",
  "shouldSetTextContent",
  "createInstance",
  "funtion  combo_box_create( parent, x, y, w, h ) :null, 0, 0, 0, 0",
  "funtion  widget_set_self_layout_params( widget, x, y, w, h ) :undefined, 50, 50, 168, 30",
  "funtion  widget_set_name( widget, name ) :undefined, zlg",
  "funtion  combo_box_set_options( widget, options ) :undefined, 0.Zzzzzzzz;1.Lzzzzzzz;2.Gzzzzzzz",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_VALUE_CHANGED, function (evt) {
        var target = src_1.TEvent.cast(evt).target;
        var instance = src_1.TComboBox.cast(target);
        console.log(instance.name + ' changed: ' + (instance.value));
        return TRet.OK;
    }, null",
  "finalizeInitialChildren",
  "getChildHostContext",
  "shouldSetTextContent",
  "createInstance",
  "funtion  combo_box_create( parent, x, y, w, h ) :null, 0, 0, 0, 0",
  "funtion  widget_set_self_layout_params( widget, x, y, w, h ) :undefined, 10, 90, 168, 30",
  "funtion  widget_set_name( widget, name ) :undefined, awtk",
  "funtion  combo_box_set_options( widget, options ) :undefined, ",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_VALUE_CHANGED, function (evt) {
        var target = src_1.TEvent.cast(evt).target;
        var instance = src_1.TComboBox.cast(target);
        console.log(instance.name + ' changed: ' + (instance.value));
        return TRet.OK;
    }, null",
  "finalizeInitialChildren",
  "createInstance",
  "funtion  window_create( parent, x, y, w, h ) :null, 0, 0, 0, 0",
  "appendInitialChild",
  "appendInitialChild",
  "appendInitialChild",
  "finalizeInitialChildren",
  "prepareForCommit",
  "appendNodeChild",
  "resetAfterCommit",
  "getPublicInstance",
  "getPublicInstance",
  "funtion  widget_add_child( widget, child ) :undefined, [object Object]",
  "funtion  widget_add_child( widget, child ) :undefined, [object Object]",
  "funtion  widget_add_child( widget, child ) :undefined, [object Object]",
  "funtion  widget_layout( widget ) :undefined",
  "win1 子控件 数量: 3",
  "funtion  combo_box_append_option( widget, value, text ) :undefined, 10, Aaaaaaa",
  "funtion  combo_box_append_option( widget, value, text ) :undefined, 20, Waaaaaa",
  "funtion  combo_box_append_option( widget, value, text ) :undefined, 30, Taaaaaa",
  "funtion  combo_box_append_option( widget, value, text ) :undefined, 40, Kaaaaaa",
  "funtion  combo_box_set_selected_index( widget, index ) :undefined, 3",
]
`;
